src/bin/entry_completion.rs

```diff
@@ -0,0 +1,110 @@
+extern crate glib;
+extern crate gdk;
+extern crate gio;
+extern crate gtk;
+
+use gio::prelude::*;
+use gtk::prelude::*;
+
+use std::env::args;
+
+macro_rules! clone {
+    (@param _) => ( _ );
+    (@param $x:ident) => ( $x );
+    ($($n:ident),+ => move |$($p:tt),*| $body:expr) => (
+        {
+            $( let $n = $n.clone(); )+
+                move |$(clone!(@param $p),)*| $body
+        }
+    );
+}
+
+struct Data {
+    description: String,
+}
+
+fn create_list_model() -> gtk::ListStore {
+    let col_types: [gtk::Type; 1] = [
+        gtk::Type::String,
+    ];
+
+    let data: [Data; 4] = [
+        Data {description: "France".to_string()},
+        Data {description: "Italy".to_string()},
+        Data {description: "Siberia".to_string()},
+        Data {description: "Switzerland".to_string()},
+    ];
+    let store = gtk::ListStore::new(&col_types);
+    let col_indices: [u32; 1] = [0];
+    for (_, d) in data.iter().enumerate() {
```

Comment:
Why use `enumerate()` if you ignore the index it generates anyway?

Reply:
copy-pasting like there's not tomorrow :100: 

fixed!